### ISSUE TITLE
BTC/LTC failed to send max amount because the output value didn't subtract fee since version 3.2.0

### ISSUE THREAD

**Issue by 09ae9c** [trustwallet/wallet-core#3273]
**Describe the bug**
upgrade to wallet-core 3.2.0, create a BTC or LTC transfer transaction with max amount, the output value didn't subtract fee, then this transaction will be failed.

**To Reproduce**
code like this: 
```
val signingInputBuilder = Bitcoin.SigningInput.newBuilder()
    .setAmount(sendAmount.toLong())
    .setHashType(hashType)
    .setToAddress(toAddress)
    .setChangeAddress(fromAddress)
    .setByteFee(suggestionsByteFee)
    .setUseMaxAmount(true)
    .setCoinType(config.coinType.value())
    
utxos.forEach { utxo ->
    signingInputBuilder.addPrivateKey(ByteString.copyFrom(Numeric.hexStringToByteArray(privateKey)))
    val sequence = Long.MAX_VALUE.toInt()
    val hash = ByteString.copyFrom(Numeric.hexStringToByteArray(utxo.transaction_hash).reversed().toByteArray())
    val outPoint = Bitcoin.OutPoint.newBuilder()
         .setHash(hash)
         .setIndex(utxo.index.toInt())
         .setSequence(sequence)
         .build()
    val script = ByteString.copyFrom(Numeric.hexStringToByteArray(utxo.script_hex))
    val unspentTransaction = Bitcoin.UnspentTransaction.newBuilder()
         .setOutPoint(outPoint)
         .setAmount(utxo.value.toLong())
         .setScript(script)
         .build()
     signingInputBuilder.addUtxo(unspentTransaction)
}
signingInputBuilder.plan = AnySigner.plan(signingInputBuilder.build(), config.coinType, Bitcoin.TransactionPlan.parser())
val output = AnySigner.sign(signingInputBuilder.build(), config.coinType, Bitcoin.SigningOutput.parser())
```

**the same code have different outputs in version 3.2.0 and 3.1.x**

In version 3.2.0, log for `plan` and `output.transaction` is:
```
plan.fee is 110, plan.availableAmount is 1998163, plan.amount is 1998053, plan.change is 0
transaction is:
# wallet.core.jni.proto.Bitcoin$Transaction@f6842697
inputs {
 previous_output {
   hash: "}\276\t\212\364\315\301\256\274\235\244\224x\004\005\335\371\2560Q\033\316\037E\000\234\310I3\002H\366"
 }
 sequence: -1
}
outputs {
 script: "\000\024\362\265\240XFz|n\207L\325\275\221\307\237\245\314\264\304\233"
 value: 1998163
}
version: 1
```

In version 3.1.39, log for `plan` and `output.transaction` is:
```
plan.fee is 110, plan.availableAmount is 1998163, plan.amount is 1998053, plan.change is 0
transaction is:
# wallet.core.jni.proto.Bitcoin$Transaction@d904b154
inputs {
 previous_output {
   hash: "}\276\t\212\364\315\301\256\274\235\244\224x\004\005\335\371\2560Q\033\316\037E\000\234\310I3\002H\366"
 }
 sequence: -1
}
outputs {
 script: "\000\024\362\265\240XFz|n\207L\325\275\221\307\237\245\314\264\304\233"
 value: 1998053
}
version: 1
```

**Expected behavior**
As you can see the difference between version 3.2.0 and 3.1.39, `outputs.value` is `1998163` in version 3.2.0 and it makes transaction failed, `1998053` is expected

**Comment by satoshiotomakan** [comment]
@09ae9c thank you for opening the issue!
I'm working on the fix

**Comment by Milerius** [comment]
Fixed by #3278

**Comment by satoshiotomakan** [comment]
@09ae9c could you please test if the issue has gone?
